### PR TITLE
Adding VpnSiteLink and VpnSiteLinkConnection properties.

### PR DESCRIPTION
**Pull request by nalinrajgupta**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/examples/VpnGatewayGet.json [hunk 1]
```diff
@@ -28,15 +28,58 @@
...
+                      "sharedKey": "key",
+                      "ingressBytesTransferred": 0,
+                      "egressBytesTransferred": 0,
+                      "enableBgp": false,
+                      "enableRateLimiting": false,
+                      "useLocalAzureIpAddress": false,
+                      "usePolicyBasedTrafficSelectors": false,
+                      "routingWeight": 0
+                    },
+                    "type": "Microsoft.Network/vpnGateways/vpnConnections/VpnSiteLinkConnections"
```
Comment (EvgenyAgafonchikov) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/examples/VpnGatewayGet.json]: `type` is not present in specs. Please make examples to correlate with real server responses and the adjust specs to match with them.
Please check the same in all Get/List operations.

File: specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/examples/VpnGatewayGet.json [hunk 2]
```diff
@@ -28,15 +28,58 @@
...
+                  }
+                ],
+                "type": "Microsoft.Network/vpnGateways/vpnConnections"
```
Comment (EvgenyAgafonchikov) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/examples/VpnGatewayGet.json]: same as above for `type`

File: specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/examples/VpnGatewayGet.json [hunk 3]
```diff
@@ -28,15 +28,58 @@
...
+                      "sharedKey": "key",
+                      "ingressBytesTransferred": 0,
+                      "egressBytesTransferred": 0,
+                      "enableBgp": false,
+                      "enableRateLimiting": false,
+                      "useLocalAzureIpAddress": false,
+                      "usePolicyBasedTrafficSelectors": false,
+                      "routingWeight": 0
+                    },
+                    "type": "Microsoft.Network/vpnGateways/vpnConnections/VpnSiteLinkConnections"
```
Comment (EvgenyAgafonchikov) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/examples/VpnGatewayGet.json]: same as above for `type`

File: specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/examples/VpnSiteLinkConnectionList.json [hunk 4]
```diff
@@ -0,0 +1,37 @@
+{
+  "parameters": {
+    "gatewayName": "gateway1",
+    "api-version": "2019-06-01",
+    "resourceGroupName": "rg1",
```
Comment (EvgenyAgafonchikov) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/examples/VpnSiteLinkConnectionList.json]: `connectionName` is required there

**Review by EvgenyAgafonchikov — COMMENTED** [review]
Left some comments.
Please uses [these tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) to validate things are fixed locally

File: specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/examples/VpnConnectionPut.json [hunk 5]
```diff
@@ -10,55 +10,104 @@
         "remoteVpnSite": {
           "id": "/subscriptions/subid/resourceGroups/rg1/providers/Microsoft.Network/vpnSites/vpnSite1"
         },
-        "vpnConnectionProtocolType": "IKEv1",
-        "sharedKey": "key"
+		"vpnLinkConnections": [
```
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/examples/VpnConnectionPut.json]: Please avoid using tabs in JSON files

File: specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/examples/VpnSiteLinkList.json [hunk 6]
```diff
@@ -0,0 +1,49 @@
+{
```
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/examples/VpnSiteLinkList.json]: This example file isn't referenced by the specs

**Review by anton-evseev — CHANGES_REQUESTED** [review]
Please avoid using tabs in example files for he sake of consistency across all JSON files

File: specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/examples/VpnConnectionPut.json [hunk 7]
```diff
@@ -10,55 +10,104 @@
...
+						"usePolicyBasedTrafficSelectors": false,
+						"routingWeight": 0
+					},
+					"type": "Microsoft.Network/vpnGateways/vpnConnections/VpnSiteLinkConnections"
+				}
+			],
+			 "type": "Microsoft.Network/vpnGateways/vpnConnections"
+        }
       }
-    }
```
[the remark this answers is not in the document]
Comment (nalinrajgupta) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/examples/VpnConnectionPut.json]: Fixed

File: specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/virtualWan.json [hunk 8]
```diff
@@ -2882,6 +3179,116 @@
         }
       }
     },
+    "VpnSiteLinkConnectionProperties": {
+      "properties": {
+        "VpnSiteLink": {
```
[the remark this answers is not in the document]
Comment (nalinrajgupta) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/virtualWan.json]: fixed

File: specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/examples/VpnSiteLinkList.json (hunk 6 above)
Comment (nalinrajgupta) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/examples/VpnSiteLinkList.json]: removed

File: specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/examples/VpnGatewayGet.json (hunk 1 above)
Comment (nalinrajgupta) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/examples/VpnGatewayGet.json]: type is referred using ./network.json#/definitions/Resource

File: specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/examples/VpnGatewayGet.json (hunk 3 above)
Comment (nalinrajgupta) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/examples/VpnGatewayGet.json]: type is referred using ./network.json#/definitions/Resource

File: specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/examples/VpnGatewayGet.json (hunk 2 above)
Comment (nalinrajgupta) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/examples/VpnGatewayGet.json]: type is referred using ./network.json#/definitions/Resource

File: specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/examples/VpnSitePut.json [hunk 9]
```diff
@@ -10,22 +10,32 @@
...
           ]
         },
-        "bgpProperties": {
-          "bgpPeeringAddress": "192.168.0.0",
-          "asn": 1234
-        }
+        "isSecuritySite": false,
+        "vpnSiteLinks": [
+          {
+            "name": "vpnSiteLink1",
```
[the remark this answers is not in the document]
Comment (nalinrajgupta) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/examples/VpnSitePut.json]: VpnSiteLink is a child resource and name is required here in params.

**Review by anton-evseev — APPROVED** [review]

**Comment by jhendrixMSFT** [comment]
@AutorestCI regenerate azure-sdk-for-go

**Comment by MikhailTryakhov** [comment]
Please merge it ASAP to be in the release

**Comment by MikhailTryakhov** [comment]
@nalinrajgupta FYI

**Review by EvgenyAgafonchikov — APPROVED** [review]

**Comment by number213** [comment]
@sanjaiganesh @KrisBash please review

**Comment by number213** [comment]
@Azure/arm-api-review-board for ARM review

File: specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/virtualWan.json [hunk 10]
```diff
@@ -3482,6 +3889,34 @@
         }
       },
       "description": "Vpn Profile Response for package generation."
+    },
+    "VpnLinkBgpSettings": {
+      "properties": {
+        "asn": {
```
Comment (ryansbenson) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/virtualWan.json]: Will these acronyms be known to your users?

Comment (nalinrajgupta) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/virtualWan.json]: Yes, ASN is a well known term for BGP.

File: specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/virtualWan.json [hunk 11]
```diff
@@ -2882,6 +3179,116 @@
...
+          "format": "int64",
+          "readOnly": true,
+          "description": "Egress bytes transferred."
+        },
+        "connectionBandwidth": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Expected bandwidth in MBPS."
+        },
+        "sharedKey": {
```
[the remark this answers is not in the document]
Comment (nalinrajgupta) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/virtualWan.json]: Yes, this is a secret ! Name need to be sharedKey as that is what customer is familiar with.

File: specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/virtualWan.json [hunk 12]
```diff
@@ -2574,6 +2803,67 @@
...
+          "description": "Gets a unique read-only string that changes whenever the resource is updated."
+        },
+        "name": {
+          "type": "string",
+          "description": "The name of the resource that is unique within a resource group. This name can be used to access the resource."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./network.json#/definitions/SubResource"
```
[the remark this answers is not in the document]
Comment (nalinrajgupta) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/virtualWan.json]: We have been following a convention to use subResource  for defining child resources. VpnSiteLink is a child resource of VpnSite and VpnSiteLinkConnection is a child resource of VpnConnection.

File: specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/virtualWan.json [hunk 13]
```diff
@@ -2882,6 +3179,116 @@
...
+        },
+        "etag": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Gets a unique read-only string that changes whenever the resource is updated."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./network.json#/definitions/SubResource"
```
[the remark this answers is not in the document]
Comment (nalinrajgupta) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/virtualWan.json]: We have been following a convention to use subResource  for defining child resources. VpnSiteLink is a child resource of VpnSite and VpnSiteLinkConnection is a child resource of VpnConnection.

File: specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/virtualWan.json (hunk 12 above)
[the remark this answers is not in the document]
Comment (ryansbenson) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/virtualWan.json]: It appears you do! The main concern here is your definition for all these resources doesn't contain a 'type' property which is mandatory for all resources in ARM

File: specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/virtualWan.json (hunk 13 above)
[the remark this answers is not in the document]
Comment (ryansbenson) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/virtualWan.json]: please add 'type' to all the resource that derive from SubResource

File: specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/virtualWan.json (hunk 11 above)
[the remark this answers is not in the document]
Comment (ryansbenson) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/virtualWan.json]: It looks like your samples contain the sharedKey in the get response. Is this intended?

File: specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/virtualWan.json (hunk 10 above)
Comment (ryansbenson) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/virtualWan.json]: thanks for confirming

File: specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/virtualWan.json (hunk 12 above)
[the remark this answers is not in the document]
Comment (nalinrajgupta) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/virtualWan.json]: Created a new type ChildResource

File: specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/network.json [hunk 14]
```diff
@@ -143,6 +143,21 @@
...
+          "type": "string",
+          "description": "Resource ID."
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource type."
+        }
+      },
+      "description": "Reference to another childresource.",
```
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/network.json]: `childresource` -> `child resource`

File: specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/network.json [hunk 15]
```diff
@@ -143,6 +143,21 @@
       "description": "Reference to another subresource.",
       "x-ms-azure-resource": true
     },
+    "ChildResource": {
+      "properties": {
+        "id": {
+          "type": "string",
```
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/network.json]: `id` should be read-only, I think

File: specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/network.json [hunk 16]
```diff
@@ -143,6 +143,21 @@
       "description": "Reference to another subresource.",
       "x-ms-azure-resource": true
     },
+    "ChildResource": {
```
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/network.json]: I think all child resources also have `name` and `etag` properties

**Review by anton-evseev — CHANGES_REQUESTED** [review]

File: specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/network.json (hunk 14 above)
Comment (nalinrajgupta) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/network.json]: Removed this type completely and added type property in specific resource.

File: specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/network.json (hunk 16 above)
Comment (nalinrajgupta) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/network.json]: Removed ChildResource type.

File: specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/network.json (hunk 15 above)
Comment (nalinrajgupta) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/network.json]: It wasn't readonly for existing network child resource, so I kept it same. I removed child resources completely.

**Comment by MikhailTryakhov** [comment]
hi guys,
need  to close it today due to release dates. please escalate fixes soon

**Review by ryansbenson — APPROVED** [review]
Approved from ARMs side

**Review by sergey-shandar — APPROVED** [review]

**Pull request merged**